modules/datahandling/userrequests.go

```diff
@@ -234,9 +235,48 @@ func (f userProjectsRequest) process(db dbfs.DBFS) ([]dhClosure, error) {
 			Status: messages.StatusPartialfail,
 			Tag:    f.Tag,
 			Data: struct {
-				Projects []dbfs.ProjectMeta
+				Projects []projectLookupResult
 			}{
-				Projects: projects,
+				Projects: nil,
+			},
+		}.Wrap()
+		return []dhClosure{toSenderClosure{msg: res}}, err
+	}
+
+	resultData := make([]projectLookupResult, len(projects))
+	var errOut error
+	i := 0
+	for _, p := range projects {
+		hasPermission, err := dbfs.PermissionAtLeast(f.SenderID, p.ProjectID, "read", db)
```

Comment:
this logic of permission at least and has permission can all be dropped here. it was needed in project lookup because we're uncertain if the user has permission to see it, but these projectID's are known to be the ones the user has access to, so we can skip this step